ownCloudAppShared/Client/Collection Views/View Controllers/ClientItemViewController.swift

```diff
@@ -0,0 +1,729 @@
+					if let drive = itemRecord?.item as? OCDrive {
+						if drive.identifier == itemControllerContext.drive?.identifier {
+							return true
+						}
+					}
+
+					return false
+				})
+
+				// Create combined data source from drive + additional items
+				driveSectionDataSource = OCDataSourceComposition(sources: [ singleDriveDatasource!, driveAdditionalItemsDataSource ])
+
+				// Create drive section from combined data source
+				driveSection = CollectionViewSection(identifier: "drive", dataSource: driveSectionDataSource, cellStyle: .init(with: .header), cellLayout: .list(appearance: .plain))
+			}
+
+			itemSectionDataSource = OCDataSourceComposition(sources: [itemsLeadInDataSource, queryResultsDatasource, itemsTrailingDataSource])
+			itemSection = CollectionViewSection(identifier: "items", dataSource: itemSectionDataSource, cellLayout: .list(appearance: .plain, contentInsets: NSDirectionalEdgeInsets(top: 0, leading: 0, bottom: 0, trailing: 0)), clientContext: itemControllerContext)
+
+			if let driveSection = driveSection {
+				sections.append(driveSection)
+			}
+
+			if let queryItemDataSourceSection = itemSection {
+				sections.append(queryItemDataSourceSection)
+			}
+		}
+
+		emptySection = CollectionViewSection(identifier: "empty", dataSource: emptyItemListDataSource, cellStyle: .init(with: .fillSpace), cellLayout: .fullWidth(itemHeightDimension: .estimated(54), groupHeightDimension: .estimated(54), edgeSpacing: NSCollectionLayoutEdgeSpacing(leading: .fixed(0), top: .fixed(10), trailing: .fixed(0), bottom: .fixed(10)), contentInsets: NSDirectionalEdgeInsets(top: 10, leading: 20, bottom: 10, trailing: 20)), clientContext: itemControllerContext)
+		sections.append(emptySection!)
+
+		super.init(context: itemControllerContext, sections: sections, useStackViewRoot: true, highlightItemReference: highlightItemReference)
+
+		// Track query state and recompute content state when it changes
+		stateObservation = itemsQueryDataSource?.observe(\OCDataSource.state, options: [], changeHandler: { [weak self] query, change in
+			self?.recomputeContentState()
+		})
+
+		queryRootItemObservation = query?.observe(\OCQuery.rootItem, options: [], changeHandler: { [weak self] query, change in
+			OnMainThread(inline: true) {
+				self?.clientContext?.rootItem = query.rootItem
+			}
+			self?.recomputeContentState()
+		})
+
+		// Subscribe to singleDriveDatasource for changes, to update driveSectionDataSource
+		singleDriveDatasourceSubscription = singleDriveDatasource?.subscribe(updateHandler: { [weak self] (subscription) in
+			self?.updateAdditionalDriveItems(from: subscription)
+		}, on: .main, trackDifferences: true, performIntialUpdate: true)
+
+		if let queryDatasource = query?.queryResultsDataSource {
+			emptyItemListItem = OCDataItemPresentable(reference: "_emptyItemList" as NSString, originalDataItemType: nil, version: nil)
+			emptyItemListItem?.title = "This folder is empty. Fill it with content:".localized
```

Comment:
String is missing in `Localizable.strings`

Reply:
Thanks, I added that string to the `Localizable.strings`.